hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -199,7 +199,7 @@ private void queueAssign(MasterProcedureEnv env, RegionStateNode regionNode)
 
   private void openRegion(MasterProcedureEnv env, RegionStateNode regionNode) throws IOException {
     ServerName loc = regionNode.getRegionLocation();
-    if (loc == null) {
+    if (loc == null || env.getMasterServices().getLoadBalancer().isBogusServerName(loc)) {
```

Comment:
I think we should directly check with `BOGUS_SERVER_NAME.equals(loc)` here. `BOGUS_SERVER_NAME` is anyways public.

Reply:
Good point, will fix, thanks.